h3/src/lib.rs

```diff
@@ -12,21 +12,37 @@ pub use error::Error;
 pub use proto::headers::Protocol;
 
 mod buf;
+
+#[cfg(feature="allow_access_to_core")]
```

Comment:
This is my attempt to make it hard to expose the core modules that webtransport need, let me know what you think

Reply:
Perhaps the conventional export, and then a separate module, called `internals` which is conditonal and re-exports all private imports

Reply:
I like this!!! what do you think @seanmonstar ?